src/config/loadGitHubAppsFromEnvironment.ts

```diff
@@ -1,42 +1,45 @@
-import { AppAuthStrategyOptions } from '@/types';
+import {
+  AppAuthStrategyOptions,
+  GitHubIssuesSomeoneElseCaresAbout,
+} from '@/types';
 
 export class GitHubApp {
+  num: number;
   org: string;
   auth: AppAuthStrategyOptions;
-  project;
+  project: GitHubIssuesSomeoneElseCaresAbout;
 
-  constructor(org: string, opts) {
-    this.org = org;
-    this.auth = {
-      appId: Number(opts.GH_APP_IDENTIFIER),
-      privateKey: opts.GH_APP_SECRET_KEY.replace(/\\n/g, '\n'),
-    };
-    this.project = {
-      node_id: opts.ISSUES_PROJECT_NODE_ID || 'PVT_kwDOABVQ184AOGW8',
-      product_area_field_id:
-        opts.PRODUCT_AREA_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgJEBno',
-      status_field_id: opts.STATUS_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgI_7g0',
-      response_due_date_field_id:
-        opts.RESPONSE_DUE_DATE_FIELD_ID || 'PVTF_lADOABVQ184AOGW8zgLLxGg',
-    };
+  constructor(obj) {
+    this.num = obj.num;
+    this.org = obj.org;
+    this.auth = obj.auth;
+    this.project = obj.project;
   }
 }
 
-export class GitHubAppsRegistry {
-  apps: Map<string, GitHubApp>;
+class GitHubAppsConfigHelper {
+  configs: Map<number, any>; // much looser typing than apps
 
   constructor() {
-    this.apps = new Map<string, GitHubApp>();
+    this.configs = new Map<number, object>();
   }
 
-  register(app) {
-    this.apps.set(app.org, app);
+  forNumber(num: number): object | undefined {
+    if (!this.configs.has(num)) {
+      this.configs.set(num, { num: num });
```

Comment:
I'm a bit confused about this, could you elaborate why this is setting an object here? Is it because the object returned is by reference?

Reply:
maybe function name could be more descriptive?

Reply:
This is a pretty normal pattern for populating a cache, no? We're filing configs away according to their number. (The number will eventually come from envvars like `GH_APP_1_FOO`, see the description on #482 and the implementation in #524. For now on this PR there's only one config (using the old/existing envvars) and it's hardwired to "1" below.) The `{ num: num }` is essentially a constructor for a new config object.

Reply:
I guess what is unclear to me is why `{ num: num }` is needed instead of just the empty object `{}`. If we have the config num to lookup, do we need that as part of the config later on?

Reply:
Ah. I store it on the config for convenience in debugging, it's not strictly necessary. Hopefully the refactor in 219157b307014f98e6715aa4de02615d656a84be makes it less distracting.

Reply:
Like, I've found by experience that having the thing something is indexed by also stored on the thing is a good habit, because when you have the thing and want to know what it is stored under ... well! ;-)